src/page-multi-facility/SyncReferenceFacilityModal.tsx

```diff
@@ -0,0 +1,161 @@
+import ModalDialog from "../design-system/ModalDialog";
+import { getFacilityById, useFacilities } from "../facilities-context";
+import { ReferenceFacility } from "./types";
+
+const ModalContent = styled.div`
+  display: flex;
+  flex-direction: column;
+  font-weight: normal;
+`;
+
+const SaveButton = styled(StyledButton)`
+  font-size: 14px;
+  font-weight: normal;
+`;
+
+const CancelButton = styled(StyledButton)`
+  background: transparent;
+  color: ${Colors.forest};
+  font-family: "Libre Franklin";
+  font-size: 14px;
+  line-height: 26px;
+`;
+
+const ModalFooter = styled.div`
+  display: flex;
+  flex-flow: row nowrap;
+  justify-content: space-between;
+  margin-top: 20px;
+`;
+
+const SyncDataTitleContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+const TitleText = styled.h1`
+  font-weight: 500;
+  font-size: 15px;
+  line-height: 22px;
+  color: ${Colors.forest};
+  font-family: "Libre Franklin";
+`;
+
+const Row = styled.div<{ selected: boolean }>`
+  border-bottom: 1px solid ${Colors.paleJade20};
+  color: ${(props) => (props.selected ? Colors.forest : Colors.paleJade)};
+  cursor: pointer;
+  display: flex;
+  flex-flow: row nowrap;
+  font-family: "Libre Franklin", sans serif;
+  font-size: 15px;
+  line-height: 16px;
+  letter-spacing: -0.01em;
+  padding: 24px 0;
+`;
+
+const RowIcon = styled.img`
+  margin-right: 12px;
+`;
+
+const SyncDataTitle = (
+  <SyncDataTitleContainer>
+    <TitleText>Sync Data</TitleText>
+    <TitleText>
+      To sync with daily public facility data, please select the corresponding
+      facility.
+    </TitleText>
+  </SyncDataTitleContainer>
+);
+
+interface Props {
+  facilityId: string | null;
+}
+
+interface ReferenceFacilityProps {
+  referenceFacility: ReferenceFacility;
+  onClick: () => void;
+  selected: boolean;
+}
+
+const ReferenceFacilityRow: React.FC<ReferenceFacilityProps> = ({
+  referenceFacility,
+  onClick,
+  selected,
+}) => {
+  return (
+    <Row selected={selected} onClick={onClick}>
+      <RowIcon
+        src={selected ? dataSyncSelectedIcon : dataSyncIcon}
+        alt="data sync icon"
+      />
+      {referenceFacility.canonicalName}
+    </Row>
+  );
+};
+
+const SyncReferenceFacilityModal: React.FC<Props> = ({ facilityId }) => {
+  const {
+    state: { facilities, referenceFacilities },
+    actions: { deselectFacility },
+  } = useFacilities();
+  const [selectedRefFacility, setSelectedRefFacility] = useState<string | null>(
+    null,
+  );
+  const facility = getFacilityById(facilities, facilityId);
+
+  function handleClick(refFacility: ReferenceFacility) {
+    if (selectedRefFacility !== refFacility.id) {
+      setSelectedRefFacility(refFacility.id);
+    } else {
+      setSelectedRefFacility(null);
+    }
+  }
+
+  function handleSave() {
+    if (selectedRefFacility) {
+      deselectFacility();
+      // save a synced composite facility?
+      console.log({ facility, selectedRefFacility });
```

Comment:
I think whatever happens with the saved facility data will be handled within the Facility context; you shouldn't have to interact with that mechanism directly here. All you should have to do, as you said, is update the mapping. (that should just be a new property of the scenario, both you and @sokunthel will need it for your features but it's a simple object (as described in [this section of the design doc](https://docs.google.com/document/d/1mvuQpXMPf3V164D4B7waCRArONw9iOWvE1g2SsLh03A/edit#heading=h.3ez6its1yt01)) so it shouldn't cause major merge conflicts or anything

Reply:
oh actually one thing I would add about this -- at some point we are going to have to cut the database over from the test collection to the real collection, which will invalidate any of these mappings we have saved in the meantime. I was thinking we might also want to use a "test" version of the mapping so that we can just change the key at the same time as we change the collection ID and give everyone a fresh mapping. (we can discuss more if that needs clarification)

Reply:
Oh okay, do you mean add a test property that we can then drop from the scenario document? Something like:
```
scenario: {
  ...other scenario properties...
  testReferenceFacilityMapping: {
    [facilityId]: referencefacilityId
  }
}
```

Reply:
yeah exactly